[2024-11-18T17:38:36.043+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-11-18T17:38:36.506+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_dag.spark_task scheduled__2024-05-06T00:00:00+00:00 [queued]>
[2024-11-18T17:38:36.671+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_dag.spark_task scheduled__2024-05-06T00:00:00+00:00 [queued]>
[2024-11-18T17:38:36.683+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-11-18T17:38:36.988+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): spark_task> on 2024-05-06 00:00:00+00:00
[2024-11-18T17:38:37.032+0000] {standard_task_runner.py:64} INFO - Started process 949 to run task
[2024-11-18T17:38:37.147+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spark_dag', 'spark_task', 'scheduled__2024-05-06T00:00:00+00:00', '--job-id', '629', '--raw', '--subdir', 'DAGS_FOLDER/spark_dag.py', '--cfg-path', '/tmp/tmp76w4qlvk']
[2024-11-18T17:38:37.235+0000] {standard_task_runner.py:91} INFO - Job 629: Subtask spark_task
[2024-11-18T17:38:37.570+0000] {task_command.py:426} INFO - Running <TaskInstance: spark_dag.spark_task scheduled__2024-05-06T00:00:00+00:00 [running]> on host 39153155bde1
[2024-11-18T17:38:39.487+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_dag' AIRFLOW_CTX_TASK_ID='spark_task' AIRFLOW_CTX_EXECUTION_DATE='2024-05-06T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-05-06T00:00:00+00:00'
[2024-11-18T17:38:39.553+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-11-18T17:38:39.684+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2024-11-18T17:38:39.687+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --name arrow-spark --deploy-mode client /opt/***/dags/spark_job.py
[2024-11-18T17:39:25.303+0000] {job.py:218} ERROR - Job heartbeat got an exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3371, in _wrap_pool_connect
    return fn()
           ^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 327, in connect
    return _ConnectionFairy._checkout(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 894, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 493, in checkout
    rec = pool._do_get()
          ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
    return self._create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 273, in _create_connection
    return _ConnectionRecord(self)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 388, in __init__
    self.__connect()
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 690, in __connect
    with util.safe_reraise():
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 686, in __connect
    self.dbapi_connection = connection = pool._invoke_creator(self)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/create.py", line 574, in connect
    return dialect.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 598, in connect
    return self.dbapi.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.OperationalError: could not translate host name "postgres" to address: Temporary failure in name resolution


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/jobs/job.py", line 192, in heartbeat
    self._merge_from(Job._fetch_from_db(self, session))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/api_internal/internal_api_call.py", line 115, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/retries.py", line 89, in wrapped_function
    for attempt in run_with_db_retries(max_retries=retries, logger=logger, **retry_kwargs):
  File "/home/airflow/.local/lib/python3.11/site-packages/tenacity/__init__.py", line 443, in __iter__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/tenacity/__init__.py", line 418, in exc_check
    raise retry_exc.reraise()
          ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/tenacity/__init__.py", line 185, in reraise
    raise self.last_attempt.result()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/retries.py", line 98, in wrapped_function
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/jobs/job.py", line 316, in _fetch_from_db
    session.merge(job)
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 3056, in merge
    return self._merge(
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 3136, in _merge
    merged = self.get(
             ^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2853, in get
    return self._get_impl(
           ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2975, in _get_impl
    return db_load_fn(
           ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/loading.py", line 530, in load_on_pk_identity
    session.execute(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1716, in execute
    conn = self._connection_for_bind(bind)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1555, in _connection_for_bind
    return self._transaction._connection_for_bind(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 750, in _connection_for_bind
    conn = bind.connect()
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/future/engine.py", line 412, in connect
    return super(Engine, self).connect()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3325, in connect
    return self._connection_cls(self, close_with_result=close_with_result)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 96, in __init__
    else engine.raw_connection()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3404, in raw_connection
    return self._wrap_pool_connect(self.pool.connect, _connection)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3374, in _wrap_pool_connect
    Connection._handle_dbapi_exception_noconnection(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2208, in _handle_dbapi_exception_noconnection
    util.raise_(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 3371, in _wrap_pool_connect
    return fn()
           ^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 327, in connect
    return _ConnectionFairy._checkout(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 894, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 493, in checkout
    rec = pool._do_get()
          ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
    return self._create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 273, in _create_connection
    return _ConnectionRecord(self)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 388, in __init__
    self.__connect()
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 690, in __connect
    with util.safe_reraise():
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 686, in __connect
    self.dbapi_connection = connection = pool._invoke_creator(self)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/create.py", line 574, in connect
    return dialect.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 598, in connect
    return self.dbapi.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "postgres" to address: Temporary failure in name resolution

(Background on this error at: https://sqlalche.me/e/14/e3q8)
[2024-11-18T17:40:33.005+0000] {job.py:226} ERROR - Job heartbeat failed with error. Scheduler is in unhealthy state
[2024-11-18T17:41:20.757+0000] {job.py:226} ERROR - Job heartbeat failed with error. Scheduler is in unhealthy state
[2024-11-18T17:42:14.866+0000] {job.py:226} ERROR - Job heartbeat failed with error. Scheduler is in unhealthy state
[2024-11-18T17:42:37.102+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO SparkContext: Running Spark version 3.5.3
[2024-11-18T17:42:37.108+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO SparkContext: OS info Linux, 5.15.153.1-microsoft-standard-WSL2, amd64
[2024-11-18T17:42:37.110+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO SparkContext: Java version 17.0.12
[2024-11-18T17:42:37.360+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-11-18T17:42:37.719+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO ResourceUtils: ==============================================================
[2024-11-18T17:42:37.721+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-11-18T17:42:37.723+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO ResourceUtils: ==============================================================
[2024-11-18T17:42:37.740+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO SparkContext: Submitted application: Airflow Spark Job
[2024-11-18T17:42:37.832+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-11-18T17:42:37.851+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO ResourceProfile: Limiting resource is cpu
[2024-11-18T17:42:37.854+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-11-18T17:42:37.998+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:37 INFO SecurityManager: Changing view acls to: ***
[2024-11-18T17:42:38.018+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:38 INFO SecurityManager: Changing modify acls to: ***
[2024-11-18T17:42:38.028+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:38 INFO SecurityManager: Changing view acls groups to:
[2024-11-18T17:42:38.030+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:38 INFO SecurityManager: Changing modify acls groups to:
[2024-11-18T17:42:38.033+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:38 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-11-18T17:42:38.789+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:38 INFO Utils: Successfully started service 'sparkDriver' on port 34413.
[2024-11-18T17:42:38.891+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:38 INFO SparkEnv: Registering MapOutputTracker
[2024-11-18T17:42:39.002+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO SparkEnv: Registering BlockManagerMaster
[2024-11-18T17:42:39.051+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-11-18T17:42:39.057+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-11-18T17:42:39.089+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-11-18T17:42:39.156+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-c8e20933-6751-470f-9bb8-ebffe2f83de0
[2024-11-18T17:42:39.186+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-11-18T17:42:39.218+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-11-18T17:42:39.404+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-11-18T17:42:39.654+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-11-18T17:42:39.659+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
[2024-11-18T17:42:39.681+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO Utils: Successfully started service 'SparkUI' on port 4042.
[2024-11-18T17:42:39.994+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:39 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark:7077...
[2024-11-18T17:42:40.233+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO TransportClientFactory: Successfully created connection to spark/172.19.0.4:7077 after 143 ms (0 ms spent in bootstraps)
[2024-11-18T17:42:40.595+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241118174240-0002
[2024-11-18T17:42:40.632+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 36977.
[2024-11-18T17:42:40.633+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO NettyBlockTransferService: Server created on 39153155bde1:36977
[2024-11-18T17:42:40.636+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-11-18T17:42:40.658+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 39153155bde1, 36977, None)
[2024-11-18T17:42:40.666+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO BlockManagerMasterEndpoint: Registering block manager 39153155bde1:36977 with 434.4 MiB RAM, BlockManagerId(driver, 39153155bde1, 36977, None)
[2024-11-18T17:42:40.672+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 39153155bde1, 36977, None)
[2024-11-18T17:42:40.682+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:40 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 39153155bde1, 36977, None)
[2024-11-18T17:42:41.496+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:41 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-11-18T17:42:43.127+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:43 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-11-18T17:42:43.144+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:43 INFO SharedState: Warehouse path is 'file:/app/spark-warehouse'.
[2024-11-18T17:42:53.639+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:53 INFO CodeGenerator: Code generated in 1482.608324 ms
[2024-11-18T17:42:53.937+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:53 INFO SparkContext: Starting job: showString at NativeMethodAccessorImpl.java:0
[2024-11-18T17:42:54.114+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO DAGScheduler: Got job 0 (showString at NativeMethodAccessorImpl.java:0) with 1 output partitions
[2024-11-18T17:42:54.121+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO DAGScheduler: Final stage: ResultStage 0 (showString at NativeMethodAccessorImpl.java:0)
[2024-11-18T17:42:54.138+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO DAGScheduler: Parents of final stage: List()
[2024-11-18T17:42:54.153+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO DAGScheduler: Missing parents: List()
[2024-11-18T17:42:54.220+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[6] at showString at NativeMethodAccessorImpl.java:0), which has no missing parents
[2024-11-18T17:42:54.608+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 13.1 KiB, free 434.4 MiB)
[2024-11-18T17:42:54.708+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 6.8 KiB, free 434.4 MiB)
[2024-11-18T17:42:54.729+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 39153155bde1:36977 (size: 6.8 KiB, free: 434.4 MiB)
[2024-11-18T17:42:54.738+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO SparkContext: Created broadcast 0 from broadcast at DAGScheduler.scala:1585
[2024-11-18T17:42:54.773+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[6] at showString at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0))
[2024-11-18T17:42:54.777+0000] {spark_submit.py:634} INFO - 24/11/18 17:42:54 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks resource profile 0
[2024-11-18T17:43:00.436+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:00 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241118174240-0002/0 on worker-20241118173202-172.19.0.5-46113 (172.19.0.5:46113) with 1 core(s)
[2024-11-18T17:43:00.449+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:00 INFO StandaloneSchedulerBackend: Granted executor ID app-20241118174240-0002/0 on hostPort 172.19.0.5:46113 with 1 core(s), 1024.0 MiB RAM
[2024-11-18T17:43:00.451+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:00 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241118174240-0002/1 on worker-20241118173203-172.19.0.6-45105 (172.19.0.6:45105) with 1 core(s)
[2024-11-18T17:43:00.457+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:00 INFO StandaloneSchedulerBackend: Granted executor ID app-20241118174240-0002/1 on hostPort 172.19.0.6:45105 with 1 core(s), 1024.0 MiB RAM
[2024-11-18T17:43:00.864+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:00 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241118174240-0002/0 is now RUNNING
[2024-11-18T17:43:00.930+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:00 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241118174240-0002/1 is now RUNNING
[2024-11-18T17:43:05.310+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.19.0.6:52402) with ID 1,  ResourceProfileId 0
[2024-11-18T17:43:05.391+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.19.0.5:55834) with ID 0,  ResourceProfileId 0
[2024-11-18T17:43:05.445+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO BlockManagerMasterEndpoint: Registering block manager 172.19.0.6:39645 with 434.4 MiB RAM, BlockManagerId(1, 172.19.0.6, 39645, None)
[2024-11-18T17:43:05.502+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO BlockManagerMasterEndpoint: Registering block manager 172.19.0.5:46507 with 434.4 MiB RAM, BlockManagerId(0, 172.19.0.5, 46507, None)
[2024-11-18T17:43:05.567+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0) (172.19.0.6, executor 1, partition 0, PROCESS_LOCAL, 9122 bytes)
[2024-11-18T17:43:05.712+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (172.19.0.6 executor 1): java.io.InvalidClassException: org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095
[2024-11-18T17:43:05.712+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:597)
[2024-11-18T17:43:05.713+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:05.713+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:05.713+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:05.714+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:05.714+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2224)
[2024-11-18T17:43:05.714+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1733)
[2024-11-18T17:43:05.715+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:509)
[2024-11-18T17:43:05.715+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:467)
[2024-11-18T17:43:05.715+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:87)
[2024-11-18T17:43:05.715+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:129)
[2024-11-18T17:43:05.716+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:579)
[2024-11-18T17:43:05.716+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-11-18T17:43:05.716+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-11-18T17:43:05.717+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-11-18T17:43:05.717+0000] {spark_submit.py:634} INFO - 
[2024-11-18T17:43:05.717+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSetManager: Starting task 0.1 in stage 0.0 (TID 1) (172.19.0.6, executor 1, partition 0, PROCESS_LOCAL, 9122 bytes)
[2024-11-18T17:43:05.736+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSetManager: Lost task 0.1 in stage 0.0 (TID 1) on 172.19.0.6, executor 1: java.io.InvalidClassException (org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095) [duplicate 1]
[2024-11-18T17:43:05.738+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSetManager: Starting task 0.2 in stage 0.0 (TID 2) (172.19.0.5, executor 0, partition 0, PROCESS_LOCAL, 9122 bytes)
[2024-11-18T17:43:05.811+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSetManager: Lost task 0.2 in stage 0.0 (TID 2) on 172.19.0.5, executor 0: java.io.InvalidClassException (org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095) [duplicate 2]
[2024-11-18T17:43:05.814+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSetManager: Starting task 0.3 in stage 0.0 (TID 3) (172.19.0.6, executor 1, partition 0, PROCESS_LOCAL, 9122 bytes)
[2024-11-18T17:43:05.830+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSetManager: Lost task 0.3 in stage 0.0 (TID 3) on 172.19.0.6, executor 1: java.io.InvalidClassException (org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095) [duplicate 3]
[2024-11-18T17:43:05.832+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 ERROR TaskSetManager: Task 0 in stage 0.0 failed 4 times; aborting job
[2024-11-18T17:43:05.837+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool
[2024-11-18T17:43:05.839+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSchedulerImpl: Cancelling stage 0
[2024-11-18T17:43:05.839+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO TaskSchedulerImpl: Killing all running tasks in stage 0: Stage cancelled: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.19.0.6 executor 1): java.io.InvalidClassException: org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095
[2024-11-18T17:43:05.840+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:597)
[2024-11-18T17:43:05.840+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:05.841+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:05.841+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:05.841+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:05.842+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2224)
[2024-11-18T17:43:05.842+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1733)
[2024-11-18T17:43:05.842+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:509)
[2024-11-18T17:43:05.843+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:467)
[2024-11-18T17:43:05.843+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:87)
[2024-11-18T17:43:05.843+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:129)
[2024-11-18T17:43:05.843+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:579)
[2024-11-18T17:43:05.844+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-11-18T17:43:05.844+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-11-18T17:43:05.844+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-11-18T17:43:05.845+0000] {spark_submit.py:634} INFO - 
[2024-11-18T17:43:05.845+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-11-18T17:43:05.845+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO DAGScheduler: ResultStage 0 (showString at NativeMethodAccessorImpl.java:0) failed in 11.494 s due to Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.19.0.6 executor 1): java.io.InvalidClassException: org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095
[2024-11-18T17:43:05.846+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:597)
[2024-11-18T17:43:05.846+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:05.847+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:05.847+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:05.847+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:05.848+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2224)
[2024-11-18T17:43:05.848+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1733)
[2024-11-18T17:43:05.848+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:509)
[2024-11-18T17:43:05.849+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:467)
[2024-11-18T17:43:05.850+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:87)
[2024-11-18T17:43:05.850+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:129)
[2024-11-18T17:43:05.851+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:579)
[2024-11-18T17:43:05.851+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-11-18T17:43:05.851+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-11-18T17:43:05.852+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-11-18T17:43:05.852+0000] {spark_submit.py:634} INFO - 
[2024-11-18T17:43:05.853+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-11-18T17:43:05.853+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:05 INFO DAGScheduler: Job 0 failed: showString at NativeMethodAccessorImpl.java:0, took 11.909008 s
[2024-11-18T17:43:06.032+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-11-18T17:43:06.033+0000] {spark_submit.py:634} INFO - File "/opt/***/dags/spark_job.py", line 16, in <module>
[2024-11-18T17:43:06.034+0000] {spark_submit.py:634} INFO - df.show()
[2024-11-18T17:43:06.035+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 947, in show
[2024-11-18T17:43:06.035+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 965, in _show_string
[2024-11-18T17:43:06.035+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-11-18T17:43:06.036+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-11-18T17:43:06.036+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2024-11-18T17:43:06.039+0000] {spark_submit.py:634} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o41.showString.
[2024-11-18T17:43:06.039+0000] {spark_submit.py:634} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.19.0.6 executor 1): java.io.InvalidClassException: org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095
[2024-11-18T17:43:06.040+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:597)
[2024-11-18T17:43:06.040+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:06.041+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:06.041+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:06.041+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:06.041+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2224)
[2024-11-18T17:43:06.042+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1733)
[2024-11-18T17:43:06.042+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:509)
[2024-11-18T17:43:06.042+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:467)
[2024-11-18T17:43:06.042+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:87)
[2024-11-18T17:43:06.042+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:129)
[2024-11-18T17:43:06.043+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:579)
[2024-11-18T17:43:06.043+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-11-18T17:43:06.043+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-11-18T17:43:06.043+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-11-18T17:43:06.044+0000] {spark_submit.py:634} INFO - 
[2024-11-18T17:43:06.044+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-11-18T17:43:06.044+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)
[2024-11-18T17:43:06.044+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)
[2024-11-18T17:43:06.044+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)
[2024-11-18T17:43:06.045+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-11-18T17:43:06.045+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-11-18T17:43:06.046+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-11-18T17:43:06.046+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)
[2024-11-18T17:43:06.047+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)
[2024-11-18T17:43:06.047+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)
[2024-11-18T17:43:06.047+0000] {spark_submit.py:634} INFO - at scala.Option.foreach(Option.scala:407)
[2024-11-18T17:43:06.048+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)
[2024-11-18T17:43:06.048+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)
[2024-11-18T17:43:06.048+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
[2024-11-18T17:43:06.048+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
[2024-11-18T17:43:06.049+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2024-11-18T17:43:06.049+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)
[2024-11-18T17:43:06.049+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)
[2024-11-18T17:43:06.049+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2414)
[2024-11-18T17:43:06.050+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2433)
[2024-11-18T17:43:06.050+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:530)
[2024-11-18T17:43:06.052+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:483)
[2024-11-18T17:43:06.052+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.CollectLimitExec.executeCollect(limit.scala:61)
[2024-11-18T17:43:06.052+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.collectFromPlan(Dataset.scala:4333)
[2024-11-18T17:43:06.053+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$head$1(Dataset.scala:3316)
[2024-11-18T17:43:06.053+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4323)
[2024-11-18T17:43:06.053+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:546)
[2024-11-18T17:43:06.053+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4321)
[2024-11-18T17:43:06.054+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-11-18T17:43:06.054+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-11-18T17:43:06.054+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-11-18T17:43:06.054+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-11-18T17:43:06.055+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-11-18T17:43:06.055+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4321)
[2024-11-18T17:43:06.055+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.head(Dataset.scala:3316)
[2024-11-18T17:43:06.056+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.take(Dataset.scala:3539)
[2024-11-18T17:43:06.056+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.getRows(Dataset.scala:280)
[2024-11-18T17:43:06.056+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.showString(Dataset.scala:315)
[2024-11-18T17:43:06.056+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-11-18T17:43:06.056+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-11-18T17:43:06.057+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-11-18T17:43:06.057+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-11-18T17:43:06.057+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-11-18T17:43:06.058+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-11-18T17:43:06.058+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-11-18T17:43:06.058+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-11-18T17:43:06.058+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-11-18T17:43:06.059+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-11-18T17:43:06.059+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-11-18T17:43:06.059+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-11-18T17:43:06.059+0000] {spark_submit.py:634} INFO - Caused by: java.io.InvalidClassException: org.apache.spark.scheduler.Task; local class incompatible: stream classdesc serialVersionUID = -6188971942555435033, local class serialVersionUID = 553100815431272095
[2024-11-18T17:43:06.060+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:597)
[2024-11-18T17:43:06.060+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:06.060+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:06.060+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2051)
[2024-11-18T17:43:06.061+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1898)
[2024-11-18T17:43:06.061+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2224)
[2024-11-18T17:43:06.061+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1733)
[2024-11-18T17:43:06.061+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:509)
[2024-11-18T17:43:06.061+0000] {spark_submit.py:634} INFO - at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:467)
[2024-11-18T17:43:06.062+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:87)
[2024-11-18T17:43:06.062+0000] {spark_submit.py:634} INFO - at org.apache.spark.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:129)
[2024-11-18T17:43:06.062+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:579)
[2024-11-18T17:43:06.062+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-11-18T17:43:06.062+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-11-18T17:43:06.063+0000] {spark_submit.py:634} INFO - ... 1 more
[2024-11-18T17:43:06.063+0000] {spark_submit.py:634} INFO - 
[2024-11-18T17:43:06.179+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO SparkContext: Invoking stop() from shutdown hook
[2024-11-18T17:43:06.180+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-11-18T17:43:06.190+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO SparkUI: Stopped Spark web UI at http://39153155bde1:4042
[2024-11-18T17:43:06.193+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-11-18T17:43:06.194+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-11-18T17:43:06.226+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-11-18T17:43:06.278+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO MemoryStore: MemoryStore cleared
[2024-11-18T17:43:06.281+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO BlockManager: BlockManager stopped
[2024-11-18T17:43:06.286+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-11-18T17:43:06.290+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-11-18T17:43:06.302+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO SparkContext: Successfully stopped SparkContext
[2024-11-18T17:43:06.303+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO ShutdownHookManager: Shutdown hook called
[2024-11-18T17:43:06.304+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO ShutdownHookManager: Deleting directory /tmp/spark-54946fe0-1e48-41b9-ac99-b0cf090fc4c9
[2024-11-18T17:43:06.308+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO ShutdownHookManager: Deleting directory /tmp/spark-5834b825-c7e5-4fdc-b2ab-2dd966ce9bdc/pyspark-f735333f-398a-4a9d-8eec-8d096c980b81
[2024-11-18T17:43:06.315+0000] {spark_submit.py:634} INFO - 24/11/18 17:43:06 INFO ShutdownHookManager: Deleting directory /tmp/spark-5834b825-c7e5-4fdc-b2ab-2dd966ce9bdc
[2024-11-18T17:43:06.416+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-11-18T17:43:06.434+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --name arrow-spark --deploy-mode client /opt/***/dags/spark_job.py. Error code is: 1.
[2024-11-18T17:43:06.440+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_dag, task_id=spark_task, run_id=scheduled__2024-05-06T00:00:00+00:00, execution_date=20240506T000000, start_date=20241118T173836, end_date=20241118T174306
[2024-11-18T17:43:06.462+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 629 for task spark_task (Cannot execute: spark-submit --master spark://spark:7077 --name arrow-spark --deploy-mode client /opt/***/dags/spark_job.py. Error code is: 1.; 949)
[2024-11-18T17:43:06.490+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2024-11-18T17:43:06.517+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-18T17:43:06.524+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
